contract/propertyMarketplace.sol

```diff
@@ -57,22 +74,56 @@ contract PropertyMarketplace {
         SoldOut
     }
 
-    mapping (uint => Property) internal properties;
-
- 
+    mapping(uint => Property) internal properties;
+    mapping(uint => bool) private exists;
 
+    /// @dev checks if a property exist
+    modifier exist(uint _index) {
+        require(exists[_index], "Query of non existent property");
```

Comment:
nothing is passed to the exist variable in this PR so maybe it would be better to check if the index is in the properties mapping? or am I missing something?

Reply:
Ok i just sure the exist function but not sure if I agree/understand that choice for the implementation, would love to hear the rationale

Reply:
Oh I'm sorry, I forgot to add the related line when writing a property. In your case, your functions have enough checks to prevent the bugs I could find that were related to that, but the transactions would be reverted with a misleading reason.